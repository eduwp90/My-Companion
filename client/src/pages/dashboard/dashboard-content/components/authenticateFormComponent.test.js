import { render, screen } from '@testing-library/react';
import AuthenticateFormComponent from './authenticateFormComponent';
import Auth from '../../../../helpers/auth';


let error = {
  isError: false,
  errorMessage: '',
};

let event = {
  email: '',
  password: '',
}

const handleChange = e => {
  1+1;
};

const invalidPassword = event.password !== event.repeat;

const invalidEmail = event.email.length
  ? !Auth.validateEmail(event.email)
  : false;


describe ('BasicInfoFormComponent', () => {
  test ('BasicInfoFormComponent should match login snapshot', () => {
    const { container } = render(<AuthenticateFormComponent error={error} event={event} handleChange={handleChange} isRegistered={true} invalidPassword={undefined} invalidEmail={undefined} spacing={"8"} />);
    expect (container.firstChild).toMatchSnapshot();
  })

  test ('BasicInfoFormComponent should match register snapshot', () => {
    const { container } = render(<AuthenticateFormComponent error={error} event={event} handleChange={handleChange} isRegistered={false} invalidPassword={invalidPassword} invalidEmail={invalidEmail} spacing={"3"} />);
    expect (container.firstChild).toMatchSnapshot();
  })

  test('should render password and email fields if logging in', () => {
    render(<AuthenticateFormComponent error={error} event={event} handleChange={handleChange} isRegistered={true} invalidPassword={undefined} invalidEmail={undefined} spacing={"8"} />)
    screen.getByPlaceholderText(/Enter email/)
    screen.getByPlaceholderText(/Enter password/)
    const secondPasswordInput = screen.queryByPlaceholderText(/Repeat password/)
    expect(secondPasswordInput).not.toBeInTheDocument()
  })

  test('should render password (x2) and email fields if registering', () => {
    render(<AuthenticateFormComponent error={error} event={event} handleChange={handleChange} isRegistered={false} invalidPassword={invalidPassword} invalidEmail={invalidEmail} spacing={"3"} />)
    screen.getByPlaceholderText(/Enter email/)
    screen.getByPlaceholderText(/Enter password/)
    screen.getByPlaceholderText(/Repeat password/)
  })

  test('should render email error message when registering with invalid email but matching passwords', () => {
    render(<AuthenticateFormComponent error={error} event={event} handleChange={handleChange} isRegistered={false} invalidPassword={false} invalidEmail={true} spacing={"3"} />)
    screen.getByText(/Should be a valid email./)
    const invalidPasswordError = screen.queryByText(/Passwords should be the equal./)
    expect(invalidPasswordError).not.toBeInTheDocument()
  })

  test('should render password error message when registering with unmatching passwords but correct email', () => {
    render(<AuthenticateFormComponent error={error} event={event} handleChange={handleChange} isRegistered={false} invalidPassword={true} invalidEmail={false} spacing={"3"} />)
    screen.getByText(/Passwords should be the equal./)
    const invalidEmailError = screen.queryByText(/Should be a valid email./)
    expect(invalidEmailError).not.toBeInTheDocument()
  })

  test('should render email error message only when registering for the first time', () => {
    render(<AuthenticateFormComponent error={error} event={event} handleChange={handleChange} isRegistered={true} invalidPassword={false} invalidEmail={true} spacing={"3"} />)
    const invalidEmailError = screen.queryByText(/Should be a valid email./)
    expect(invalidEmailError).not.toBeInTheDocument()
  })

})