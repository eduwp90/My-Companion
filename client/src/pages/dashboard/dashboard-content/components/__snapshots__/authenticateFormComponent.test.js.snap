// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicInfoFormComponent BasicInfoFormComponent should match login snapshot 1`] = `
<div
  class="chakra-stack css-1clhiew"
>
  <div
    class="chakra-form-control css-0"
    role="group"
  >
    <div
      class="chakra-input__group css-4302v8"
    >
      <div
        class="chakra-input__left-element css-1u2k0bl"
      >
        <svg
          class="chakra-icon css-2u1jce"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12,.5A11.634,11.634,0,0,0,.262,12,11.634,11.634,0,0,0,12,23.5a11.836,11.836,0,0,0,6.624-2,1.25,1.25,0,1,0-1.393-2.076A9.34,9.34,0,0,1,12,21a9.132,9.132,0,0,1-9.238-9A9.132,9.132,0,0,1,12,3a9.132,9.132,0,0,1,9.238,9v.891a1.943,1.943,0,0,1-3.884,0V12A5.355,5.355,0,1,0,12,17.261a5.376,5.376,0,0,0,3.861-1.634,4.438,4.438,0,0,0,7.877-2.736V12A11.634,11.634,0,0,0,12,.5Zm0,14.261A2.763,2.763,0,1,1,14.854,12,2.812,2.812,0,0,1,12,14.761Z"
            fill="currentColor"
          />
        </svg>
      </div>
      <input
        aria-required="true"
        class="chakra-input css-0"
        id="email"
        placeholder="Enter email"
        required=""
        type="email"
      />
    </div>
  </div>
  <div
    class="chakra-form-control css-0"
    role="group"
  >
    <div
      class="chakra-input__group css-4302v8"
    >
      <div
        class="chakra-input__left-element css-1u2k0bl"
      >
        <svg
          class="chakra-icon css-2u1jce"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19.5,9.5h-.75V6.75a6.75,6.75,0,0,0-13.5,0V9.5H4.5a2,2,0,0,0-2,2V22a2,2,0,0,0,2,2h15a2,2,0,0,0,2-2V11.5A2,2,0,0,0,19.5,9.5Zm-9.5,6a2,2,0,1,1,3,1.723V19.5a1,1,0,0,1-2,0V17.223A1.994,1.994,0,0,1,10,15.5ZM7.75,6.75a4.25,4.25,0,0,1,8.5,0V9a.5.5,0,0,1-.5.5H8.25a.5.5,0,0,1-.5-.5Z"
            fill="currentColor"
          />
        </svg>
      </div>
      <input
        aria-required="true"
        autocomplete="off"
        class="chakra-input css-0"
        id="password"
        placeholder="Enter password"
        required=""
        type="password"
      />
    </div>
  </div>
</div>
`;

exports[`BasicInfoFormComponent BasicInfoFormComponent should match register snapshot 1`] = `
<div
  class="chakra-stack css-15duo8n"
>
  <div
    class="chakra-form-control css-0"
    role="group"
  >
    <div
      class="chakra-input__group css-4302v8"
    >
      <input
        aria-required="true"
        class="chakra-input css-0"
        id="email"
        placeholder="Enter email"
        required=""
        type="email"
      />
    </div>
  </div>
  <div
    class="chakra-form-control css-0"
    role="group"
  >
    <div
      class="chakra-input__group css-4302v8"
    >
      <input
        aria-required="true"
        autocomplete="off"
        class="chakra-input css-0"
        id="password"
        placeholder="Enter password"
        required=""
        type="password"
      />
    </div>
  </div>
  <div
    class="chakra-form-control css-0"
    role="group"
  >
    <div
      class="chakra-input__group css-4302v8"
    >
      <input
        aria-required="true"
        autocomplete="off"
        class="chakra-input css-0"
        id="repeat"
        placeholder="Repeat password"
        required=""
        type="password"
      />
    </div>
  </div>
</div>
`;
